### PR TITLE
Mentor profile page

### PR DESCRIPTION
**Pull request by roarosa**
Added full mentor profile view functionality to wishlist, search, and homepage. Contact information is visible for assigned mentors in homepage and hidden for search and wishlist.

### CODE REVIEW

File: db.php [hunk 1]
```diff
@@ -289,8 +289,7 @@ function chooseMentor() {
 
 
 	 function getMentor($mentor) {
-		global $_USER;
-		$user = $_USER['uid'];
+		$user = $mentor;
```
Comment (jdreddaway) [inline db.php:292]: This means that anyone can access anyone's mentor, right? Is that ok?

If that's a problem, we should probably just make a note that it needs to be fixed.

File: js/controllers.js [hunk 2]
```diff
@@ -202,21 +188,25 @@ appControllers.controller('HomeController', ['$scope', '$http', '$location', fun
...
+        async: true,
+        success: function(result) {
+          $scope.show_identifier = true;
+          $scope.myMentor = result[0];
+          console.log("getMentor");
+          $scope.$apply();
+        },
+        type: 'GET',
+        error: $scope.ajaxError
+        // error: ajaxError
```
Comment (jdreddaway) [inline js/controllers.js]: Delete this comment?

File: db.php (hunk 1 above)
Comment (roarosa) [inline db.php:292]: Previously "getMentor" was actually not returning the user's mentor, it was just returning the logged in user. This change does mean that you can use getMentor to get the information for any mentor, but it's actually necessary to get the data for wishlist/search previews too. If we want to change it we'll have to make a second api call. One to get the full information for a currently matched mentor, one to get partial information for any mentor.

File: js/controllers.js (hunk 2 above)
Comment (roarosa) [inline js/controllers.js]: Huh?

File: db.php (hunk 1 above)
Comment (jdreddaway) [inline db.php:292]: Whoops I misread what this does. I thought it got the specific user's mentor. This is actually getting the info for any mentor. I think it's fine the way it is.

File: js/controllers.js (hunk 2 above)
Comment (jdreddaway) [inline js/controllers.js]: Just being nitpicky, but the `// error: ajaxError` isn't necessary, right?

Comment (roarosa) [inline js/controllers.js]: Ah, yes. Remnants from previous code. Let me go in and delete it really quick.

Comment (roarosa) [inline js/controllers.js]: Deleted that line in new commit. In a later pull request we may be able to go in and clean up similar comments in other areas.

**Pull request merged**